oop-static_modifier-medium-q5 (#696)
canvas-gamification/canvas-gamification-junit-tests: src/oop/static_modifier/medium/q5/QuestionText.md

# oop-static_modifier-medium-q5

- Question Type: Java

## Question Text

> As the manager of a local apparel store, you are tasked with modelling your store's selection. Make a Java program
> called **Hat**  to model the hats you have in the store using the requirements below:

Reviewer: ```suggestion
As the manager of a local apparel store, you are tasked with modeling your store's selection. Write a Java program
called `Hat`  to model the hats you have in the store using the requirements below. Then write a class called `Test` to test `Hat` based on the second set of requirements below.
```

### Hat Class

> Your `Hat` class already the following items:

Reviewer: ```suggestion
Your `Hat` class already has the following items:
```

- Attributes:

>     - A `String` attributes called **colour** that cannot be accessed in other classes

Reviewer: ```suggestion
    - A `String` attribute called **colour** that cannot be accessed in other classes
```

- Constructors:
    - A constructor that accepts a `String` and initializes the **colour** attribute.
    - This constructor is accessible from other classes.

You need to add the following items:

- Attributes:
    - A `String` attribute called **companyName** that has its value, initially `Nike`, shared by all **Hat** instances.

>     - Note that all attributes in this class should be private

Reviewer 2: Just add to the attribute description that it should not be visible to other classes. Having this note doesn't really make sense when there is only one attribute they are adding.

> - Methods:

Reviewer: make a note of visibility

>     - a `toString` method which returns a String of the information about the `Hat` class of the form
>       `This is a x Hat that is y in colour` where x is the value of **companyName** attribute and y is the
>       value of **colour** attribute

Reviewer: ```suggestion
    - a `toString` method which returns a String of the information about the `Hat` class in the form
      `This is a x Hat that is y in colour` where x is the value of the **companyName** attribute and y is the
      value of the **colour** attribute
```

### Test Class:

> Your `Test` class will have a main method which initializes a `Hat` of **colour** value of `"Green"`. Then,

Reviewer: ```suggestion
Your `Test` class should have a main method which initializes a `Hat` object of **colour** value of `"Green"`. Then,
```

> print the toString for this class.

Reviewer: ```suggestion
print the object on its own line using the toString method.
```

# oop-static_modifier-medium-q5

- Question Type: Java

## Question Text

As the manager of a local apparel store, you are tasked with modelling your store's selection. Make a Java program

> called **Hat**.

Reviewer 2: ```suggestion
called **Hat**  to model the hats you have in the store using the requirements below:
```

> called **Hat** that has a static variable called **companyLogo** that is shared by all the objects of the class.

Reviewer: get rid of all duplicate class info

### Hat Class

Your `Hat` class already the following items:

- Attributes:
    - A `String` attributes called **colour**.

>     - Note that all attributes in this class are private

Reviewer: this should not be plural

- Constructors:
    - A constructor that accepts a `String` and initializes the **colour** attribute.
    - This constructor is accessible from other classes.

You need to add the following items:

- Attributes:

>     - A `String` attribute called **companyLogo** that has its value, initially `Nike`, shared by all **Hat** instances.

Reviewer 2: Can we call this **companyName** instead of **companyLogo**?

# oop-static_modifier-medium-q5

- Question Type: Java

## Question Text

As the manager of a local apparel store, you are tasked with modeling your store's selection. Write a Java program

> called `Hat`  to model the hats you have in the store using the requirements below. Then write a class called `Test` to

Reviewer: ```suggestion
called `Hat` to model the hats you have in the store using the requirements below. Then write a class called `Test` to
```

test `Hat` based on the second set of requirements below.

### Hat Class

Your `Hat` class already has the following items:

- Attributes:
    - A `String` attribute called **colour** that cannot be accessed in other classes

- Constructors:
    - A constructor that accepts a `String` and initializes the **colour** attribute.
    - This constructor is accessible from other classes.

You need to add the following items:

- Attributes:
    - A `String` attribute called **companyName** that has its value, initially `Nike`, shared by all **Hat** instances,
      and is not visible from another class.

- Methods:
    - a `toString` method, that is visible from other classes, which returns a String of the information about the `Hat`

>       class in the form `This is a x Hat that is y in colour` where x is the value of the **companyName** attribute and

Reviewer: modify the strings elsewhere to match this format. Currently none of the others say "Hat" in them
